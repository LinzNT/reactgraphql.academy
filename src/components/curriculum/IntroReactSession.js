import React from 'react'
import Ul, { Li } from '../layout/Ul'
import Session from './Session'

const IntroReactSession = ({ hideTitle }) => (
  <Session hideTitle={hideTitle} title="Thinking in React">
    <Ul>
      <Li>The Imperative to Declarative Shift</Li>
      <Li>Componentization</Li>
      <Li>
        What is React made up of
        <Ul>
          <Li>Props & State</Li>
          <Li>One way data binding</Li>
          <Li>Components with and without state. Classes Vs. Functions</Li>
        </Ul>
      </Li>
      <Li>
        Virtual DOM and JSX
        <Ul>
          <Li>React Components, Elements, and Instances</Li>
          <Li>JSX and React.createElement</Li>
        </Ul>
      </Li>
      <Li>Developer tools</Li>
    </Ul>
  </Session>
)

export default IntroReactSession
