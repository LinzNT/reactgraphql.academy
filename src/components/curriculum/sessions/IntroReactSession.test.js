import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import IntroReactSession from './IntroReactSession'

describe('<IntroReactSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<IntroReactSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
