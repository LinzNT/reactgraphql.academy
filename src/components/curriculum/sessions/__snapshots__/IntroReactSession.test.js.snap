// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroReactSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      The Imperative to Declarative Shift
    </styled.li>
    <styled.li>
      Componentization
    </styled.li>
    <styled.li>
      What is React made up of
      <styled.ul>
        <styled.li>
          Props & State
        </styled.li>
        <styled.li>
          One way data binding
        </styled.li>
        <styled.li>
          Components with and without state. Classes Vs. Functions
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Virtual DOM and JSX
      <styled.ul>
        <styled.li>
          React Components, Elements, and Instances
        </styled.li>
        <styled.li>
          JSX and React.createElement
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Developer tools
    </styled.li>
  </styled.ul>
</Session>
`;
